src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs

```diff
@@ -53,10 +98,96 @@ private static bool NeedVtableAttribute(SyntaxNode node)
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.Run<TResult>(System.Func<System.Threading.CancellationToken, System.Threading.Tasks.Task<TResult>>)", "System.Threading.Tasks.TaskToAsyncOperationAdapter`1"},
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.FromResult<TResult>(TResult)", "System.Threading.Tasks.TaskToAsyncOperationAdapter`1" },
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.FromException<TResult>(System.Exception)", "System.Threading.Tasks.TaskToAsyncOperationAdapter`1" },
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.CanceledOperation<TResult>()", "System.Threading.Tasks.TaskToAsyncOperationAdapter`1" },
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.Run<TResult, TProgress>(System.Func<System.Threading.CancellationToken, System.IProgress<TProgress>, System.Threading.Tasks.Task<TResult>>)", "System.Threading.Tasks.TaskToAsyncOperationWithProgressAdapter`2" },
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.FromResultWithProgress<TResult, TProgress>(TResult)", "System.Threading.Tasks.TaskToAsyncOperationWithProgressAdapter`2" },
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.FromExceptionWithProgress<TResult, TProgress>(System.Exception)", "System.Threading.Tasks.TaskToAsyncOperationWithProgressAdapter`2" },
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.CanceledOperationWithProgress<TResult, TProgress>()", "System.Threading.Tasks.TaskToAsyncOperationWithProgressAdapter`2" },
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.Run<TProgress>(System.Func<System.Threading.CancellationToken, System.IProgress<TProgress>, System.Threading.Tasks.Task>)", "System.Threading.Tasks.TaskToAsyncActionWithProgressAdapter`1" },
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.CompletedActionWithProgress<TProgress>()", "System.Threading.Tasks.TaskToAsyncActionWithProgressAdapter`1" },
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.FromExceptionWithProgress<TProgress>(System.Exception)", "System.Threading.Tasks.TaskToAsyncActionWithProgressAdapter`1" },
+            { "System.Runtime.InteropServices.WindowsRuntime.AsyncInfo.CanceledActionWithProgress<TProgress>()", "System.Threading.Tasks.TaskToAsyncActionWithProgressAdapter`1" }
+        };
+
+        // There are several async operation related methods that can be called.
+        // But they are all under the AsyncInfo static class or is the AsAsyncOperation
+        // extension method.
+        static bool IsAsyncOperationMethodCall(SyntaxNode node)        
+        {
+            if (node is InvocationExpressionSyntax method)
+            {
+                var methodStr = method.ToString();

```

Comment:
This is nice, but I'm wondering whether there's a more efficient way to avoid calling `ToString()` on every single invocation just to check the name. I suppose we could manually check for `IdentifierNameSyntax` or `SimpleMemberAccessSyntax`, but it feels a bit brittle. @333fred do you know if there's a better, efficient way to get the name of the method for a given `InvocationExpressionSyntax`?

Reply:
Walk through any member access expressions to the tokens you are about. Looking at this code, you're not handling `System.Runtime.InteropServices.WindowsRuntime.AsyncInfo` formats or `global::System.Runtime.InteropServices.WindowsRuntime.AsyncInfo`. I would always recommend using a syntax visualizer here to understand the format.

Reply:
Thanks, I was able check the member accesses and get it working.